src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/CohereEmbeddingClient.java

```diff
@@ -0,0 +1,117 @@
+package io.stargate.sgv2.jsonapi.service.embedding.operation;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import io.quarkus.rest.client.reactive.QuarkusRestClientBuilder;
+import io.smallrye.mutiny.Uni;
+import jakarta.ws.rs.HeaderParam;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+import org.eclipse.microprofile.rest.client.annotation.ClientHeaderParam;
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+/**
+ * Interface that accepts a list of texts that needs to be vectorized and returns embeddings based
+ * of chosen Cohere model.
+ */
+public class CohereEmbeddingClient implements EmbeddingService {
+  private String apiKey;
+  private String modelName;
+  private String baseUrl;
+  private final CohereEmbeddingService embeddingService;
+
+  public CohereEmbeddingClient(String baseUrl, String apiKey, String modelName) {
+    this.apiKey = apiKey;
+    this.modelName = modelName;
+    this.baseUrl = baseUrl;
+    embeddingService =
+        QuarkusRestClientBuilder.newBuilder()
+            .baseUri(URI.create(baseUrl))
+            .build(CohereEmbeddingService.class);
+  }
+
+  @RegisterRestClient
+  public interface CohereEmbeddingService {
+    @POST
+    @Path("/embed")
+    @ClientHeaderParam(name = "Content-Type", value = "application/json")
+    Uni<EmbeddingResponse> embed(
+        @HeaderParam("Authorization") String accessToken, EmbeddingRequest request);
+  }
+
+  private record EmbeddingRequest(String[] texts, String model, String input_type) {}
+
+  private static class EmbeddingResponse {
+
+    public EmbeddingResponse() {}
+
+    @JsonIgnore private String id;
```

Comment:
You can add `@JsonIgnore` here, although more common is to add annotation to getter method.
(both work as long as names are linked)..

Reply:
Actually, if we do not actually need these values and just want to ignore them coming from server, can instead use `@JsonIgnoreProperties` on class declaration, and omit field/getter/setter?
I'll add note above to show how.